src/firebase_functions/private/util.py

```diff
@@ -310,6 +311,41 @@ def firebase_config() -> None | FirebaseConfig:
     return FirebaseConfig(storage_bucket=json_data.get("storageBucket"))
 
 
+def nanoseconds_timestamp_conversion(time: str) -> _dt.datetime:
+    """Converts a nanosecond timestamp and returns a datetime object of the current time in UTC"""
+
+    # Separate the date and time part from the nanoseconds.
+    datetime_str, nanosecond_str = time.replace("Z", "").split(".")
```

Comment:
Are we not handling the lowercase `z` here, similar to how it's handled in `is_precision_timestamp`?

Reply:
Found an easier way to do it 
```
datetime_str, nanosecond_str = time.replace("Z", "").replace("z", "").split(".")
```